### ISSUE TITLE
cmake build failing with CUDA not found error (I have CUDA-11.2 with VS Integration installed)

### ISSUE THREAD

**Issue by sg12056** [AlexeyAB/darknet#7836]
If something doesn’t work for you, then show 2 screenshots:
1. screenshots of your issue
![image](https://user-images.githubusercontent.com/21309171/123502979-b1574d00-d66d-11eb-8f5c-c5d1c2a4141c.png)

- VS Community 2019
- CUDA 11.2 with VS Integration (GTX 1660Ti card)
- Path variables are set as can be seen in the screenshot
- Same error when trying to build from PowerShell
![image](https://user-images.githubusercontent.com/21309171/123503067-23c82d00-d66e-11eb-88a7-fad606bcecf3.png)

If you do not get an answer for a long time, try to find the answer among Issues with a Solved label: https://github.com/AlexeyAB/darknet/issues?q=is%3Aopen+is%3Aissue+label%3ASolved

**Comment by sg12056** [comment]
I updated my CUDA to 11.3 as I thought 11.2 could have some issue, it still showed the same error message.
Then I added ```set(CMAKE_CUDA_COMPILER "C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.3/bin/nvcc.exe")``` to the CMakeLists.txt file , now the error changed to ```CUDA_ARCHITECTURES is empty for target "cmTC_062f0".```

![image](https://user-images.githubusercontent.com/21309171/123533785-dfa05f80-d735-11eb-9d5e-010d5bb483b4.png)

**Comment by yuhe0925** [comment]
I think you have to use CUDA 10.2

**Comment by sg12056** [comment]
@yuhe0925 I'll try that, but the [documentation](https://github.com/AlexeyAB/darknet#requirements-for-windows-linux-and-macos) says CUDA >= 10.2 should work.

**Comment by stephanecharette** [comment]
> @yuhe0925 I'll try that, but the [documentation](https://github.com/AlexeyAB/darknet#requirements-for-windows-linux-and-macos) says CUDA >= 10.2 should work.
> 
> EDIT : CUDA 10.2 is also giving the same error -

Correct, Darknet requires 10.2 or later.  It definitely does not require 10.2 exactly.

**Comment by HosseinMohammadii** [comment]
Hello.
Could you solve your problem?
I face the same
